src/transaction.rs

```diff
@@ -321,6 +340,24 @@ mod tests {
         assert_matches!(memfind(&tx_bytes, &tx.signature.as_ref()), Some(SIG_OFFSET));
         assert_matches!(memfind(&tx_bytes, &tx.from.as_ref()), Some(PUB_KEY_OFFSET));
     }
+    #[test]
+    fn test_userdata_layout() {
+        let mut tx = test_tx();
+        tx.userdata = Some(vec![1, 2, 3]);
+        let sign_data = tx.get_sign_data();
+        let tx_bytes = serialize(&tx).unwrap();
+        assert!(tx_bytes.len() < 256);
+        assert_matches!(memfind(&tx_bytes, &sign_data), Some(SIGNED_DATA_OFFSET));
+        assert_matches!(memfind(&tx_bytes, &tx.signature.as_ref()), Some(SIG_OFFSET));
+        assert_matches!(memfind(&tx_bytes, &tx.from.as_ref()), Some(PUB_KEY_OFFSET));
+        let tx2 = deserialize(&tx_bytes).unwrap();
```

Comment:
When a test requires two transactions, the rest of the codebase consistently uses `tx0` and `tx1`. Can you follow suit?